phpBB/includes/acp/acp_bbcodes.php

```diff
@@ -231,6 +236,11 @@ function main($id, $mode)
 						trigger_error($user->lang['BBCODE_HELPLINE_TOO_LONG'] . adm_back_link($this->u_action), E_USER_WARNING);
 					}
 
+					if (strlen($bbcode_font_icon) > 50)
```

Comment:
I'd also add a simple regex check

Reply:
Updated string length check and added `preg_match`.